wrouter/router_test.go

```diff
@@ -328,3 +330,49 @@ func mustMatchHandler(t *testing.T, method, path string, pathVars map[string]str
 		matched[fmt.Sprintf("[%s] %s", method, path)] = true
 	})
 }
+
+// Tests that RouteHandlerMiddleware are called in the right order on the right routes.
+func TestRouterMiddlewareRegistration(t *testing.T) {
+	newMarkingMiddleware := func(marking string) wrouter.RouteHandlerMiddleware {
+		return func(rw http.ResponseWriter, req *http.Request, reqVals wrouter.RequestVals, next wrouter.RouteRequestHandler) {
+			curr := req.Context().Value("marking")
+			if curr == nil {
+				req = req.WithContext(context.WithValue(req.Context(), "marking", []string{marking}))
```

Comment:
nit:
Let's pull out the `"marking"` context key into a local type preferably a `struct{}` so we pass staticcheck linting since we're providing a hardcoded string here this fails [SA1029](https://staticcheck.io/docs/checks#SA1029)

Reply:
done!